internal/providers/docker/components/container/build.go

```diff
@@ -238,3 +257,47 @@ func tarBuildContext(w io.WriteCloser, root string) (err error) {
 	}
 	return nil
 }
+
+func createIgnoreMatcher(buildContext string) (*fileutils.PatternMatcher, error) {
+	// Initialize patterns with always ignored files.
+	const ignoreBasename = ".dockerignore"
+	patterns := []string{
+		ignoreBasename,
```

Comment:
Why are we including `.dockerignore` in the list of ignored files? Docker doesn't:

```
❯ cat Dockerfile
FROM bash
COPY . .

❯ cat .dockerignore
my-file

❯ docker run $(docker build -q .) cat .dockerignore
my-file
```

Or am I misunderstanding what's happening?

Reply:
I was copying what I saw in moby/moby, but I forget where. Maybe I misread it.